java/ql/src/semmle/code/java/dataflow/NullGuards.qll

```diff
@@ -70,6 +72,11 @@ Expr clearlyNotNullExpr(Expr reason) {
     (reason = r1 or reason = r2)
   )
   or
+  exists(SwitchExpr s |
+    s = result and
+    forall(Expr resultExpr | resultExpr = s.getAResult() | resultExpr = clearlyNotNullExpr(reason))
```

Comment:
This apparently never holds, possibly due to the recursion? Reporting `result` as `reason` works though:
```suggestion
    forall(Expr resultExpr | resultExpr = s.getAResult() | resultExpr = clearlyNotNullExpr())
    and reason = result
```